### DISCUSSION TITLE
Prometheus rule still being evaluated even though defined in thanos ruler

### DISCUSSION THREAD

**Discussion by psandhu79** [prometheus-operator/prometheus-operator#3963] [Q&A]
Question regarding the Thanos ruler operator.

If I specify the rules i wish for the Thanos ruler to pick up , it looks like they are still evaluated by the Prometheus system too.

When i use the thanos query and select deduplication it selects the Prometheus recorded metric because i drop the rule_replica, If i unselect deduplication i see both results.

How do I get query to prioritise the thanos ruler recorded rule.

Pod

**Comment by simonpasquier** [comment]
:wave: you need to use different `ruleSelector` for Thanos ruler and Prometheus: for instance, Prometheus selects all rules with label `foo` while Thanos ruler selects all rules without label `foo`.

**Reply by psandhu79** [reply]
Thank you so much
